nntrainer/layers/embedding.cpp

```diff
@@ -133,10 +135,38 @@ void EmbeddingLayer::copy(std::shared_ptr<Layer> l) {
 }
 
 void EmbeddingLayer::calcDerivative() {
-  // NYI
+  return; // intended
```

Comment:
It seems like embedding layer does not pass the derivative back. If yes, then please throw here.

Reply:
No. It need to be update the gradient. so.. It is not like input layer. Input Layer does not need to do backwarding. But we do not have detail control about it. This need another change but.. for not it is ok as it is. ( You can see line 353 in neuralnet.cpp )

Reply:
This layer calculates the derivative for the next layer in backpropagation.
In order for the network to run correctly, either this layer must give back the derivative or throw to ensure that the layer behind it (if any) does not receive any garbage gradient.
Just a simple return invalidates the invariant of the network training.

Reply:
I will leave another issue for this. but not this pr.

Reply:
Please raise a issue for this :)